code/game/objects/machinery/overwatch.dm

```diff
@@ -1214,7 +1244,129 @@ GLOBAL_LIST_EMPTY(active_cas_targets)
+				if (RETREAT_ORDER)
+					send_retreat_orders(target_turf)
+					return
+		to_chat(usr, "<span class='warning'>Your last order was too recent</span>")
+		return
+	return
+
+	
+/obj/machinery/computer/camera_advanced/overwatch/proc/notify_marine(mob/living/marine, turf/target_turf) ///Send an order to that specific marine if it's on the right z level
+	if(marine.z == target_turf.z)
+		marine.playsound_local(marine, "sound/effects/CIC_order.ogg", 10, 1)
+		switch(current_order)
+			if (ATTACK_ORDER)
+				to_chat(marine,"<span class='ordercic'>Command is urging you to attack the ennemy at [target_turf.loc.name]</span>")
+				return
+			if (DEFEND_ORDER)
+				to_chat(marine,"<span class='ordercic'>Command is urging you to defend our positions in [target_turf.loc.name]</span>")
+				return
+			if (RETREAT_ORDER)
+				to_chat(marine,"<span class='ordercic'>Command is urging you retreat now from [target_turf.loc.name]</span>")
+				return
+		
+
+/obj/machinery/computer/camera_advanced/overwatch/proc/send_attack_orders(turf/target_turf) ///Send an attack order to all marines
+	new /obj/effect/temp_visual/order/attack_order(target_turf)
+	var/obj/screen/arrow/arrow_hud
+	var/datum/atom_hud/squad/squad_hud = GLOB.huds[DATA_HUD_SQUAD]
+	var/list/final_list = squad_hud.hudusers
+	final_list -= current_user //We don't want the eye to have an arrow, it's silly
+	for(var/hud_user in final_list)
+		arrow_hud = new /obj/screen/arrow/attack_order_arrow
+		arrow_hud.add_hud(hud_user, target_turf)
+		notify_marine(hud_user, target_turf)
+
+/obj/machinery/computer/camera_advanced/overwatch/proc/send_retreat_orders(turf/target_turf) ///Send a retreat order
+	new /obj/effect/temp_visual/order/retreat_order(target_turf)
+	var/datum/atom_hud/squad/squad_hud = GLOB.huds[DATA_HUD_SQUAD]
+	var/list/final_list = squad_hud.hudusers
+	final_list -= current_user
+	for(var/hud_user in final_list)
+		notify_marine(hud_user, target_turf)
+
+/obj/machinery/computer/camera_advanced/overwatch/proc/send_defend_orders(turf/target_turf) ///Send a defend order
+	new /obj/effect/temp_visual/order/defend_order(target_turf)
+	var/obj/screen/arrow/arrow_hud
+	var/datum/atom_hud/squad/squad_hud = GLOB.huds[DATA_HUD_SQUAD]
+	var/list/final_list = squad_hud.hudusers
+	final_list -= current_user
+	for(var/hud_user in squad_hud.hudusers)
+		arrow_hud = new /obj/screen/arrow/defend_order_arrow
+		arrow_hud.add_hud(hud_user, target_turf)
+		notify_marine(hud_user, target_turf)
+
+/datum/action/innate/attack_order
+	name = "Send Attack Order"
+	background_icon_state = "template2"
+	action_icon_state = "attack"
+
+/datum/action/innate/defend_order
+	name = "Send Defend Order"
+	background_icon_state = "template2"
+	action_icon_state = "defend"
+
+/datum/action/innate/retreat_order
+	name = "Send Retreat Order"
+	background_icon_state = "template2"
+	action_icon_state = "retreat"
+
+///Set the order as selected on the overwatch console
+/datum/action/innate/proc/set_selected_order(var/order_name)
```

Comment:
why is this a proc of /datum/action/innate and not of a /datum/action/innate/order, make a generic typepath for orders